@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -177,9 +221,211 @@ export class MultiPartUploadWrapper {
      * The Promise with the HTTP status or rejects in case of errors.
      */
     async upload(
+        data: string | File | Blob | ArrayBufferLike | BlobData,
         opts: UploadOptions,
         abotrSignal?: AbortSignal
     ): Promise<number> {
+        const defaultParallelRequests = 6;
+        const parallelRequests =
+            opts.parallelRequests || defaultParallelRequests;
+
+        const parts: { id: string; number: number }[] = [];
+
+        const blobData = await this.prepareBlobData(data);
+
+        const dataSize = blobData.size();
+
+        const request = await this.getUploadRequest({
+            blobVersion: opts.blobVersion,
+            catalogHrn: opts.catalogHrn,
+            abotrSignal
+        });
+
+        const startMultipartResponse = await request.startMultipartUpload({
+            contentType: opts.contentType,
+            handle: opts.handle,
+            layerId: opts.layerId,
+            billingTag: opts.billingTag,
+            contentEncoding: opts.contentEncoding
+        });
```

Comment:
Not sure about it. Users can create an instance once and set `OlpClientSettings` and use the method `upload` many times to upload different data to different catalogs/Blob storages. Or you prefer to change it and the user should create a new instance of the wrapper for each upload?

Reply:
Of course, it is only for one upload, for one catalog. How else will the status callbacks work and the total chunks to upload, if anybody is using it and it is changed all the time? This is not a magic class that solves all the problems, it is a tool for one single operation, which is to upload a blob to one catalog and one layer partition.

Reply:
events - it's just lambdas and each call of `upload` method will call its own lambda. 

But I can change the class and instead of 

```
const wrapper = new MulpipartUploadWrapper(settings);
wrapper.upload(opts1);
wrapper.upload(ops2);
... 
etc.
```

User will need to do that:

```
const wrapper = new MulpipartUploadWrapper(settings, ops1);
wrapper.upload();

const wrapper = new MulpipartUploadWrapper(settings, ops2);
wrapper.upload();
```

The behaviour will be the same

Reply:
updated. Moved the uploading options to the constructor.